### ISSUE TITLE
raise exceptions

### ISSUE THREAD

**Issue by hartsock** [vmware/pyvmomi#78]
At present actions that seem like they would throw exceptions do not. Instead the error is transmitted as part of a data object and the programmer is expected to check for an exception before proceeding.

Example:

``` python
   for vm in vmList:
       if vm.name in vmnames:
           if vm.runtime.powerState != vim.VirtualMachinePowerState.poweredOn:
              task = vm.PowerOn()
              WaitForTasks([task], si)
           lease = vm.ExportVm()
           if lease.error:
                raise lease.error
```

... this is currently _anticipated_ behavior, so most programmers who have been using the lib for a number of years will have code built to handle this kind of check. The problem is _new_ programmers will expect the method to actually raise an exception (the method's documentation states that it raises exceptions when in pyVmomi it never does).
# Enhancement

When an exception occurs on a remote method, raise the exception as if it occurred in Python. This is the most pythonic way to handle raising an exception.

**Comment by larsskj** [comment]
Yes - +1.

**Comment by lingfish** [comment]
Is this still not done?

**Comment by larsskj** [comment]
> So are you suggesting that every time a remote method on a managed object is called it checks for an "error" property and raising an exception?

Exactly: That's the pythonese way to do it. As it is today, it's forcing Python developers into writing code in a very non-pythonese way. That's a serious usability degradation.
